src/ServicePulse.Host/app/js/services/services.historyperiods.js

```diff
@@ -0,0 +1,42 @@
+; (function (window, angular, undefined) {
+    'use strict';
+
+    function service($cookies) {
+
+        var periods = [
+                { value: 1,  text: "1m",  refreshInterval:  1 * 1000 },
+                { value: 5,  text: "5m",  refreshInterval:  5 * 1000 },
+                { value: 10, text: "10m", refreshInterval: 10 * 1000 },
+                { value: 15, text: "15m", refreshInterval: 15 * 1000 },
+                { value: 30, text: "30m", refreshInterval: 30 * 1000 },
+                { value: 60, text: "1h",  refreshInterval: 60 * 1000 }
+        ];
+
+        this.saveSelectedPeriod = function(period) {
+            $cookies.put('history_period', period.value);
+        };
+
+        this.getDefaultPeriod = function () {
+            var storedPeriodValue = $cookies.get('history_period');
+            var storedPeriod = periods[periods.findIndex(function(period) {
```

Comment:
Instead of this we could have wrote:
`periods.find( (period) => period.value == storedPeriodValue); `

But we can do it in a separate PR. A bit less of code.